### PR TITLE
Update SPDX tools-golang lib to v0.5.0

### PR DESCRIPTION
**Pull request by kzantow**
This PR updates spdx/tools-golang to v0.5.0 (-rc1).

### CODE REVIEW

**Review by lumjjb — APPROVED** [review]
LGTM! Nice!

**Comment by kzantow** [comment]
I'm going to do a bit more manual testing to see if I can find any cases we didn't cover properly in tests and take it out of draft when I think things are ready.

File: syft/formats/common/spdxhelpers/to_format_model.go [hunk 1]
```diff
@@ -8,8 +8,7 @@ import (
 	"strings"
 	"time"
 
-	"github.com/spdx/tools-golang/spdx/common"
-	spdx "github.com/spdx/tools-golang/spdx/v2_3"
+	"github.com/spdx/tools-golang/spdx"
```
Comment (spiffcs) [inline syft/formats/common/spdxhelpers/to_format_model.go:11]: already so clean!

File: syft/formats/spdxtagvalue/decoder.go [hunk 2]
```diff
@@ -4,14 +4,14 @@ import (
...
-	"github.com/spdx/tools-golang/tvloader"
+	"github.com/spdx/tools-golang/tagvalue"
 
 	"github.com/anchore/syft/syft/formats/common/spdxhelpers"
 	"github.com/anchore/syft/syft/sbom"
 )
 
 func decoder(reader io.Reader) (*sbom.SBOM, error) {
-	doc, err := tvloader.Load2_3(reader)
+	doc, err := tagvalue.Read(reader)
```
Comment (spiffcs) [inline syft/formats/spdxtagvalue/decoder.go:14]: This is also awesome

**Review by spiffcs — APPROVED** [review]
All the tests pass and from my perspective most of the updates in the PR are the library changes `common ==> spdx`

LGTM

If there is other testing you want done or edge cases to call out let me know, but I think we can get this in and release
